src/lib/analyzer.ts

```diff
@@ -45,14 +45,16 @@ export class ConstructAnalyzer {
       for (const signal of pillar.signals) {
         const rawValue = packageData[signal.name];
 
-        if (rawValue === undefined) {
-          console.warn(`Signal data not found: ${signal.name}`);
-          continue;
+        let level: number = 1;
+        let points: number = 0;
+
+        if (rawValue !== undefined) {
+          level = signal.benchmarks(rawValue);
+          points = this.convertLevelToPoints(level);
+        } else {
+          console.warn(`Signal data not found: ${signal.name}, assigning score of 0`);
```

Comment:
instead of this, wouldn't it be better if your `benchmarks` and `convertLevelToPoints` functions handled `undefined` correctly?

```ts
signal.benchmarks(undefined) -> return undefined
this.convertLevelToPoints(undefined) -> return 0 (plus a warning)
```